src/dbos-executor.ts

```diff
@@ -800,19 +814,50 @@ export class DBOSExecutor implements DBOSExecutorContext {
 
       args = DBOSJSON.parse(ires.serializedInputs) as T;
       status = ires.status;
+      $deadline = ires.deadline;
       await debugTriggerPoint(DEBUG_TRIGGER_WORKFLOW_ENQUEUE);
     }
 
```

Comment:
@chuck-dbos wanted me to ensure the timeout was cleared regardless of how Promise.race completed and to wait for the cancelled callPromise to complete before returnning.